### ISSUE TITLE
StripMining breaks Vectorization

### ISSUE THREAD

**Issue by Gabbeo** [spcl/dace#283]
**Describe the bug**
Applying StripMining and then Vectorization to a map causes the loop to only calculate the first tile.

**Output**
With only Vectorization (length 4):
```
X: [0.5488135  0.71518937 0.60276338 0.54488318 0.4236548  0.64589411 0.43758721 0.891773  ]
Y: [0.96366276 0.38344152 0.79172504 0.52889492 0.56804456 0.92559664 0.07103606 0.0871293 ]
Z: [0.52887114 0.2742333  0.47722286 0.28818595 0.2406548  0.59783742 0.03108447 0.07769956]
```
With StripMining (length 4) and Vectorization (length 4):
```
X: [0.5488135  0.71518937 0.60276338 0.54488318 0.4236548  0.64589411 0.43758721 0.891773  ]
Y: [0.96366276 0.38344152 0.79172504 0.52889492 0.56804456 0.92559664 0.07103606 0.0871293 ]
Z: [0.52887114 0.2742333  0.47722286 0.28818595 0.         0.       0.         0.        ]
```

**To Reproduce**
Use the following code:
```#!/usr/bin/env python
from __future__ import print_function

import dace
import numpy as np
import sys

#####################################

N = dace.symbol('N')

@dace.program(dace.float64[N], dace.float64[N], dace.float64[N])
def multiply(X, Y, Z):
    @dace.map(_[0:N])
    def mult(i):
        x << X[i]
        y << Y[i]
        z >> Z[i]
        
        z = y * x
    
#####################################


if __name__ == "__main__":
    size = 256
    
    N.set(size)

    np.random.seed(0)
    X = np.random.rand(N.get()).astype(np.float64)
    Y = np.random.rand(N.get()).astype(np.float64)
    Z = np.zeros_like(Y)
        
    multiply(X, Y, Z)
    
    np.set_printoptions(threshold=sys.maxsize)
    print("X:",X)
    print("Y:",Y)
    print("Z:",Z)
```

**Comment by tbennun** [comment]
Thanks! It looks like it has to do with the Python exclusive ranges and an off-by-one issue. We'll take a look.

**Comment by tbennun** [comment]
This will probably be solved in PR #301. We now use the tasklet connectors and arrays for vector types, which makes everything more robust to such errors.

**Comment by tbennun** [comment]
I can confirm that it now works with #301. 
I made a test out of your code example (but changed the default vector length to 2 to support the CI): https://github.com/spcl/dace/blob/dd81a0b2dea64dd97e54ea66077286cc36be3d0e/tests/transformations/tiling_vectorization_test.py

Thanks again for reporting!
